lib/templateFactory.js

```diff
@@ -187,6 +187,64 @@ class TemplateFactory extends BaseFactory {
         return super.list(config);
     }
 
+    /**
+     * List templates and associated metrics with pagination and filter options
+     * @method listWithMetrics
+     * @param  {Object}   config                  Config object
+     * @param  {Object}   config.params           Parameters to filter on
+     * @param  {Object}   config.paginate         Pagination parameters
+     * @param  {Number}   config.paginate.count   Number of items per page
+     * @param  {Number}   config.paginate.page    Specific page of the set to return
+     * @return {Promise}
+     */
+    listWithMetrics(config) {
+        return this.list(config).then((templates) => {
+            if (templates.length === 0) {
+                return templates;
+            }
+
+            // eslint-disable-next-line global-require
+            const JobFactory = require('./jobFactory');
+            const jobFactory = JobFactory.getInstance();
+
+            // eslint-disable-next-line global-require
+            const BuildFactory = require('./buildFactory');
+            const buildFactory = BuildFactory.getInstance();
+
+            const templateIds = templates.map(t => t.id);
+
+            const listConfig = {
+                params: { templateId: templateIds },
+                readOnly: true
+            };
+
+            return jobFactory.list(listConfig).then(jobs =>
+                buildFactory.list(listConfig).then(builds =>
```

Comment:
you should do it this way
`return Promise.all([
  jobFactory.list(listConfig),
  buildFactory.list(listConfig)
]).then(([jobs, builds]) => {`

Reply:
updated, thanks